src/server.h

```diff
@@ -800,9 +800,11 @@ struct serverObject {
                               * and most significant 16 bits access time). */
     unsigned hasexpire : 1;
     unsigned hasembkey : 1;
+    unsigned hasembval : 1;
     unsigned refcount : OBJ_REFCOUNT_BITS;
-    void *ptr;
+    void *val_ptr;
```

Comment:
To make sure future contributors use objectGetVal instead of accessing this field directly, let's add a comment about it.

Idea:

```suggestion
    void *val_ptr; /* Not always present. Use objectGetVal(obj) and
                    * objectSetVal(obj, val) instead. */
```